test/unit/validator/block_builder_test.exs

```diff
@@ -30,21 +34,23 @@ defmodule Unit.Validator.ProposerTests do
     # This private key is taken from the spec test vectors
     privkey = <<0::248, 64>>
 
-    block_request = %BlockRequest{
+    block_request = %BuildBlockRequest{
       slot: pre_state.slot + 1,
       parent_root: spec_block.message.parent_root,
       proposer_index: 63,
       graffiti_message: "",
-      eth1_data: %Types.Eth1Data{
-        deposit_root: <<0::256>>,
-        deposit_count: 64,
-        block_hash: <<0::256>>
-      },
-      execution_payload: spec_block.message.body.execution_payload,
       privkey: privkey
     }
 
-    {:ok, signed_block} = Proposer.construct_block(pre_state, block_request, privkey)
+    {:ok, signed_block} =
+      private(
+        BlockBuilder.construct_block(
+          pre_state,
+          block_request,
+          spec_block.message.body.execution_payload,
+          spec_block.message.body.eth1_data
+        )
+      )
```

Comment:
We could do that, but why are you against the current approach? In general I think we should avoid making functions that are inherently private to be public just because of unit tests (we can solve it with patch)

Reply:
It's just that testing a private function _smells_, and I think that this function could be used in a _standalone_ fashion for building customized blocks (e.g. with external payload builders). But yeah, I'm probably getting ahead of myself, and we should leave this for a later refactor.